src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -154,6 +173,12 @@ void DlgPrefColors::slotApply() {
     }
 }
 
+void DlgPrefColors::slotReplaceCueColorClicked() {
+    m_pReplaceCueColorDlg->show();
+    m_pReplaceCueColorDlg->raise();
+    m_pReplaceCueColorDlg->activateWindow();
```

Comment:
Could you send the selected default color from the combobox to the color replacement dialog here?

Reply:
The applied color or the currently selected color even without hitting apply first?

Reply:
How about:
```
if selected color == old color
    load old color as color to replace
else
    load selected color as color to replace old cues with
```

Reply:
Ok this is a bit more complex than I thought because the hotcue palette could also have been changed and also auto_hotcue_colors might be on or off. Let's do this in a separate PR.

Reply:
okay, then this PR looks good to me. Just waiting on final approval from @daschuer

Reply:
While testing H ave been stumbled also over this, since the button is now well connected to the default combo box. 

How bout only allow to open the replace dialog after the user has applied all changes. Everything else has a lot of corner cases hard to model.  

And then, use always the stored default color at "too" color because this is most likely the color which should be used now. Store the default color we had after starting Mixxx as "from" color.   
If they match let the dialog decide for one other color in the current palette.

This looks like a small change that improves the usability IMHO a lot and fits in this PR.

Reply:
> How bout only allow to open the replace dialog after the user has applied all changes. 

I think that would be confusing.

Reply:
I don't mind if we force Apply or not.. 

At least the pe-selected new color should be within the palette of the palette picker.